# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: a10921d560a0
# Full Hash: a10921d560a0c2c178f98aefcd43207af8d95097
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2025-09-25 10:57 +0000
# Regressor Bug: 1980152
# File Overlap Count: 1
# Overlapping Files: dom/script/ScriptLoader.cpp
# Description:
#   Bug 1980152 - Part 1: Calculate the disk caching plan also with stencil-based requests. r=nbp
# ==============================================================================

diff -r 7adcef30c668 -r a10921d560a0 js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Thu Sep 25 10:57:33 2025 +0000
+++ b/js/src/frontend/Stencil.cpp	Thu Sep 25 10:57:33 2025 +0000
@@ -6214,3 +6214,11 @@
 
   return true;
 }
+
+JS_PUBLIC_API size_t JS::GetScriptSourceLength(JS::Stencil* stencil) {
+  const ScriptSource* source = stencil->getInitial()->source;
+  if (!source->hasSourceText()) {
+    return 0;
+  }
+  return source->length();
+}
